File: LeetCode/Python/Separate Black and White Balls.py
"""
  Separate Black and White Balls
  URL: https://leetcode.com/problems/separate-black-and-white-balls
  Difficulty: Medium
  Topic: Two Pointers, String, Greedy
"""

class Solution:
    def minimumSteps(self, s: str) -> int:
        one = 0
        result = 0
        for i in s:
            if i == '1':
                one += 1
            else:
                result += one
        return result
        

File: LeetCode/Separate Black and White Balls.swift
"""
  Separate Black and White Balls
  URL: https://leetcode.com/problems/separate-black-and-white-balls
  Difficulty: Medium
  Topic: Two Pointers, String, Greedy
"""

class Solution {
    func minimumSteps(_ s: String) -> Int {
        s.reduce(into: [0,0]) { r, c in
            if c == "1" {
                r[1] += 1
            } else {
                r[0] += r[1]
            }
        }[0]
    }
}
